tests/unit/User/AvatarUploaderTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Tests\unit\User;
+
+use Flarum\User\AvatarUploader;
+use Flarum\User\User;
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Database\Eloquent\Model;
+use Intervention\Image\ImageManagerStatic;
+use League\Flysystem\FilesystemInterface;
+use Mockery as m;
+use Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+use PHPUnit\Framework\TestCase;
+
+class AvatarUploaderTest extends TestCase
+{
+    use MockeryPHPUnitIntegration;
+
+    private $dispatcher;
+    private $filesystem;
+    private $uploader;
+
+    public function setUp()
+    {
+        $this->dispatcher = m::mock(Dispatcher::class);
+        $this->dispatcher->shouldIgnoreMissing();
+        Model::setEventDispatcher($this->dispatcher);
+
+        $this->filesystem = m::mock(FilesystemInterface::class);
+        $this->uploader = new AvatarUploader($this->filesystem);
+    }
+
+    public function test_removing_avatar_removes_file()
+    {
+        $this->filesystem->shouldReceive('has')->with('ABCDEFGHabcdefgh.png');
+        $this->filesystem->shouldReceive('delete')->with('ABCDEFGHabcdefgh.png')->andReturn(true);
```

Comment:
I don't know if this explains the issues you're having, but that return value is probably meant to belong to the `has()` mock above?

Reply:
Indded. I made a mistake here. At first I had added `andReturn` for each of them but then realized it was only really useful for `has`. And I removed the wrong one here.

I can try pulling the new master tomorrow afternoon or evening. If someone wants to tackle it before that time, feel free to do.